app/contacts/new/page.js

```diff
@@ -0,0 +1,148 @@
+'use client';
+import { useState } from "react";
+import { useRouter } from "next/navigation";
+import { Modal, Button, Form } from "react-bootstrap";
+import { contactsApi } from "@/app/components/ContactsApi";
+import Contacts from "../page";
+
+
+export default function newContactModal() {
+
+  const [firstName, setFirstName] = useState(null);
+  const [lastName, setLastName] = useState(null);
+  const [phoneNumber, setPhoneNumber] = useState(null);
+  const [email, setEmail] = useState(null);
+  const [imageUrl, setImageUrl] = useState(null);
+
+  const router = useRouter();
+
+  const handleSubmitNewContactClick = () => {
+
+    let uniqueId = Math.round(Math.random() * 100000000).toString()
+
+    let newContactData = {
+      "id": uniqueId,
+      "profile_picture": imageUrl,
+      "first_name": firstName,
+      "last_name": lastName,
+      "phone": phoneNumber,
+      "email": email
+    }
+
+    console.log(newContactData)
+
+    contactsApi.addContact(newContactData)
+
+    console.log(contactsApi.all())
+
+    router.back()
+  }
+
+  const handleClose = () => {
+    // router.push('/contacts')
+    router.back()
+  }
+
+  const formAttributes = {
+    formGroups: [
+      {
```

Comment:
DRY - a lot of repeating code. You could map an array of objects and populate this form groups